File: app/src/main/java/com/live/utils/SecureValidate.java
package com.live.utils;

import android.content.pm.PackageInfo;
import android.content.pm.PackageManager;

import com.live.LiveApp;

import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;


public class SecureValidate {

    public static String getPkgName() {
        String pkName = null;
        try {

            pkName = LiveApp.getInstance().getPackageName();
        } catch (Exception e) {
            e.printStackTrace();
        }
        return pkName;
    }

    public static int getVerCode() {
        int verCode = 0;
        try {
            String pkName = LiveApp.getInstance().getPackageName();
            verCode = LiveApp.getInstance().getPackageManager().getPackageInfo(pkName, 0).versionCode;
        } catch (Exception e) {
            e.printStackTrace();
        }
        return verCode;
    }

    public static String printSignatureMD5() {
        try {
            PackageInfo packageInfo = LiveApp.getInstance().getPackageManager().getPackageInfo(LiveApp.getInstance().getPackageName(), PackageManager.GET_SIGNATURES);

            return getMD5MessageDigest(packageInfo.signatures[0].toByteArray());

        } catch (Exception e) {
            e.printStackTrace();
        }
        return null;
    }

    private static String getMD5MessageDigest(byte[] bytes) {
        StringBuilder md5StringBuffer = new StringBuilder();
        try {
            MessageDigest messageDigest = MessageDigest.getInstance("MD5");
            messageDigest.reset();
            messageDigest.update(bytes);
            byte[] digest = messageDigest.digest();
            for (byte aDigest : digest) {
                String hexString = Integer.toHexString(aDigest & 0xff);

                if (hexString.length() == 1)
                    md5StringBuffer.append("0");

                md5StringBuffer.append(hexString);
            }
        } catch (NoSuchAlgorithmException e) {
            e.printStackTrace();
        }
        return md5StringBuffer.toString();
    }


}

File: app/src/main/java/com/live/utils/SecureValidate.kt
package com.live.utils

import android.content.pm.PackageManager.GET_SIGNATURES
import com.live.LiveApp
import java.security.MessageDigest
import java.security.NoSuchAlgorithmException


object SecureValidate {

    val pkgName: String?
        get() {
            var pkName: String? = null
            try {

                pkName = LiveApp.instance!!.packageName
            } catch (e: Exception) {
                e.printStackTrace()
            }

            return pkName
        }

    val verCode: Int
        get() {
            var verCode = 0
            try {
                val pkName = LiveApp.instance!!.packageName
                verCode = LiveApp.instance!!.packageManager.getPackageInfo(pkName, 0).versionCode
            } catch (e: Exception) {
                e.printStackTrace()
            }

            return verCode
        }

    val verName: String?
        get() {
            var verName: String? = null
            try {
                val pkName = LiveApp.instance!!.packageName
                verName = LiveApp.instance!!.packageManager.getPackageInfo(pkName, 0).versionName
            } catch (e: Exception) {
                e.printStackTrace()
            }

            return verName
        }

    fun printSignatureMD5(): String? {
        try {
            val packageInfo = LiveApp.instance!!.packageManager.getPackageInfo(LiveApp.instance!!.packageName, GET_SIGNATURES)

            return getMD5MessageDigest(packageInfo.signatures[0].toByteArray())

        } catch (e: Exception) {
            e.printStackTrace()
        }

        return null
    }

    private fun getMD5MessageDigest(bytes: ByteArray): String {
        val sb = StringBuilder()
        try {
            val messageDigest = MessageDigest.getInstance("MD5")
            val digest:ByteArray = messageDigest.digest(bytes)//对字符串加密，返回字节数组

            for (b in digest) {
                val i :Int = b.toInt() and 0xff//获取低八位有效值
                var hexString = Integer.toHexString(i)//将整数转化为16进制
                if (hexString.length < 2) {
                    hexString = "0$hexString"//如果是一位的话，补0
                }
                sb.append(hexString)
            }
            return sb.toString()
        } catch (e: NoSuchAlgorithmException) {
            e.printStackTrace()
        }
        return sb.toString()
    }


}
